### PR TITLE
Make SDK node compatible, Add login

### PR DESCRIPTION
**Pull request by Harsimran1**
- Make SDK to run in node context. Added a SessionHandler layer which gets resolved (based on 
   NODE or WEB environment) while building the SDK. 
- Webpack configuration changes to adapt to both WEB and NODE environments while building.
- New method for login in NODE environment (using client_credentials grant type)
- Code Coverage decreases a little because 'coverage' plugin shows coverage for specific environment ( If we're running tests for both the environments, coverage takes the latest report generated.)

### CODE REVIEW

File: src/templates/registrationForm.js [hunk 1]
```diff
@@ -3,8 +3,9 @@ const tagIds = {
 	zkitRegistration: 'zkitRegistration',
 };
 
-const registrationForm = document.createElement('div');
-registrationForm.innerHTML = `
+const getRegistrationForm = () => {
+	const registrationForm = document.createElement('div');
+	registrationForm.innerHTML = `
 	<div class="row">
```
Comment (Glumli) [inline src/templates/registrationForm.js]: Please indent one tab more

File: test/lib/hcRequestTest.js [hunk 2]
```diff
@@ -89,7 +89,6 @@ describe('hcRequest', () => {
 			.then((res) => {
 				expect(res.status).to.equal('201');
 				expect(getRefreshTokenStub).to.be.calledOnce;
-				expect(getAccessTokkenStub.callCount).to.equal(3);
```
Comment (Glumli) [inline test/lib/hcRequestTest.js]: Why did you remove this?

File: test/lib/sessionHandler/node.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-unused-expressions */
```
Comment (Glumli) [inline test/lib/sessionHandler/node.js]: Please nem it nodeTest.js for consitency reasons

File: test/lib/sessionHandler/node.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+
+sinonStubPromise(sinon);
+chai.use(sinonChai);
+
+const expect = chai.expect;
+
+describe('sessionHandler node', () => {
+	it('get and set sessionHandler succeeds', (done) => {
+		sessionHandler.set('HC_User', 'fakeUserId1,fakeUserAlias1');
```
Comment (Glumli) [inline test/lib/sessionHandler/node.js]: Please use testsVariables.js

File: test/lib/sessionHandler/web.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import chai from 'chai';
 import sinon from 'sinon';
```
Comment (Glumli) [inline test/lib/sessionHandler/web.js]: Please nem it webTest.js for consistency reasons

File: test/routes/authRoutesTest.js [hunk 6]
```diff
@@ -46,6 +47,76 @@ describe('user Routes', () => {
...
 	});
 
+	it('getAccessTokenFromCredentials passes', (done) => {
+		const headers = {
+			'Content-Type': 'application/x-www-form-urlencoded',
+		};
+
+
+		const params = {
+			client_id: 'fakeClientId',
```
Comment (Glumli) [inline test/routes/authRoutesTest.js]: please use testVariables.js

File: test/routes/authRoutesTest.js [hunk 7]
```diff
@@ -46,6 +47,76 @@ describe('user Routes', () => {
...
+			scope: 'user:fakeUserId',
+			grant_type: 'client_credentials',
+			client_secret: 'fakeClientSecret',
+		};
+		const credentialsStub = sinon.stub().returnsPromise().withArgs('POST', `${config.api}/auth/token`, { params, headers })
+			.resolves('pass');
+
+		authRoutes = proxyquire('../../src/routes/authRoutes', {
```
Comment (Glumli) [inline test/routes/authRoutesTest.js]: You can move the whole proxyquire stuff to beforeEach. Except of telling the stub what it expects you can then simply check with what it was called in the then block

File: test/routes/authRoutesTest.js [hunk 8]
```diff
@@ -46,6 +47,76 @@ describe('user Routes', () => {
...
+
+
+		authRoutes.getAccessTokenFromCredentials(params).then((res) => {
+			expect(res).to.equal('pass');
+			expect(credentialsStub).to.be.calledOnce;
+			expect(credentialsStub).to.be.calledWith('POST');
+			done();
+		});
+
+		requestStub.reset();
```
Comment (Glumli) [inline test/routes/authRoutesTest.js]: please reset stubs that where stubbed in beforeEach in afterEach

File: test/services/authServiceTest.js [hunk 9]
```diff
@@ -7,7 +7,7 @@ import sinonStubPromise from 'sinon-stub-promise';
 import sinonChai from 'sinon-chai';
 import config from 'config';
 
-import siriusRoutes from '../../src/routes/authRoutes';
+import authRoutes from '../../src/routes/authRoutes';
```
Comment (Glumli) [inline test/services/authServiceTest.js]: nice

File: test/services/authServiceTest.js [hunk 10]
```diff
@@ -22,7 +22,7 @@ describe('Auth', () => {
 
 	beforeEach(() => {
 		config.api = 'http://fakeUrl';
-		auth = new Auth({ clientId: '1', userId: '3213' });
+		auth = new Auth({ clientId: '1', userId: '3213', clientSecret: 'fakeClientSecret' });
```
Comment (Glumli) [inline test/services/authServiceTest.js]: Please use the testVariable.js

File: test/services/authServiceTest.js [hunk 11]
```diff
@@ -96,6 +96,7 @@ describe('Auth', () => {
 		});
 	});
 
+
```
Comment (Glumli) [inline test/services/authServiceTest.js]: Why adding another line here. Isn't one empty line enough

File: test/services/zerokitAdapterTest.js [hunk 12]
```diff
@@ -74,6 +79,10 @@ describe('zerokitAdapter', () => {
 				{ session_id: testVariables.sessionId, zerokit_id: testVariables.zeroKitId });
 		loginStub = sinon.stub()
 			.returnsPromise().resolves(testVariables.zeroKitId);
+		loginNodeStub = sinon.stub().returnsPromise();
```
Comment (Glumli) [inline test/services/zerokitAdapterTest.js]: you can simpy chain it.
```
loginNodeStub = sinon.stub()
	.returnsPromise().rejects('error')
 	.withArgs(testVariables.zeroKitId, testVariables.password)
	.resolves('success');
  ```
But my idea was to have the before each so that everything is happy and working and then if you wnt something failing you overwirte the stub in the test where it is needed

File: test/services/zerokitAdapterTest.js [hunk 13]
```diff
@@ -112,12 +121,26 @@ describe('zerokitAdapter', () => {
...
+			createTresor: createTresorStub,
+		};
+
 		zeroKitPromise = Promise.resolve(zeroKit);
 
 		zerokitAdapter = new ZerokitAdapter({ authService });
 		zerokitAdapter.zeroKit = zeroKitPromise;
 		zerokitAdapter.zKitRegistrationObject = zKitRegisterObjectPromise;
 
+		zerokitAdapterNode = new ZerokitAdapter('dummyZerokitadapter');
```
Comment (Glumli) [inline test/services/zerokitAdapterTest.js]: Just put an empty object in there

Comment (Glumli) [reply test/services/zerokitAdapterTest.js]: Or better: Your authservice
```zerokitAdapterNode.authService = {
 	clientCredentialsLogin: sinon.stub().returnsPromise().withArgs('kjhgf')
 		.resolves(''),
 };```

File: test/lib/hcRequestTest.js (hunk 2 above)
Comment (Harsimran1) [reply test/lib/hcRequestTest.js]: there are some issues with proxyquire-universal and aliases. Looking into that. Had to remove for the meantime.

File: test/services/zerokitAdapterTest.js (hunk 12 above)
Comment (Harsimran1) [reply test/services/zerokitAdapterTest.js]: Cannot chain. https://github.com/sinonjs/sinon/issues/176. Anyways moved error case in the test.

File: src/services/ZeroKitAdapter.js [hunk 14]
```diff
@@ -15,20 +16,56 @@ class ZeroKitAdapter {
...
-			script.addEventListener('load', () => {
-				if (zkit_sdk) {
-					zkit_sdk.setup(config.zkit.service_url, '');
-					resolve(zkit_sdk);
-				} else {
-					reject(new Error('Setting up ZeroKit failed.'));
-				}
+
+		if (NODE) {
+			this.zeroKit = require('zerokit-node-client')(config.zkit.service_url, 4);
```
[the remark this answers is not in the document]
Comment (Harsimran1) [reply src/services/ZeroKitAdapter.js]: This is a service version which is 4, similar to v4 `${config.zkit.service_url}/static/v4/zkit-sdk.js`  can add it to config though.

File: test/lib/hcRequestTest.js (hunk 2 above)
Comment (Harsimran1) [reply test/lib/hcRequestTest.js]: fixed the issue

File: src/HealthCloud.js [hunk 15]
```diff
@@ -30,6 +27,15 @@ class HealthCloud {
 		this.getGrantedPermissions = UserService.getGrantedPermissions.bind(UserService);
 		this.logout = zeroKitAdapter.logout.bind(zeroKitAdapter);
 
+		if (NODE) {
+			this.login = zeroKitAdapter.loginNode.bind(zeroKitAdapter);
+		}
+		if (!NODE) {
```
Comment (Glumli) [inline src/HealthCloud.js]: Why didn't you simply use else?

File: src/services/ZeroKitAdapter.js [hunk 16]
```diff
@@ -75,7 +92,14 @@ class ZeroKitAdapter {
 				return zKitLoginObject.then(loginObject => loginObject.login(zeroKitId));
 			})
 			.then(() => this.authService.idpLogin())
-			.then(() => UserService.getInternalUser())
+			.then(() => this.postLogin())
+			.then(() => ({ id: userId, alias: hcUserAlias }));
+	}
+
+
+	postLogin() {
```
Comment (Glumli) [inline src/services/ZeroKitAdapter.js]: postLogin is quite an unhappy chosen name, as It took me quite some time to figure out that you don't mean the verb.

File: test/services/zerokitAdapterTest.js [hunk 17]
```diff
@@ -95,6 +103,8 @@ describe('zerokitAdapter', () => {
 
 		authService = {
 			idpLogin: idpLoginStub,
+			clientCredentialsLogin: sinon.stub().returnsPromise().withArgs(testVariables.userId)
+				.resolves(''),
```
Comment (Glumli) [inline test/services/zerokitAdapterTest.js]: resolves doesn't need arguments

File: src/templates/loginForm.js [hunk 18]
```diff
@@ -4,20 +4,24 @@ const tagIds = {
...
+		    <button id="${tagIds.submitBtn}">ANMELDEN</button>
+			</div>
+	`;
+	return loginForm;
+};
+
 
 export { tagIds };
-export default loginForm;
+export default getLoginForm;
```
Comment (Glumli) [inline src/templates/loginForm.js]: Why did you remove the call?
Could you then also rename it to "getLoginForm" in the ZeroKitAdapter. Otherwise it looks like a variable there

File: src/HealthCloud.js (hunk 15 above)
Comment (Harsimran1) [reply src/HealthCloud.js]: makes sense

File: src/services/ZeroKitAdapter.js [hunk 19]
```diff
@@ -15,25 +16,42 @@ class ZeroKitAdapter {
...
+
+	loginNode(hcUserAlias, password) {
+		return UserService.resolveUser(hcUserAlias)
+			.then(user => this.zeroKit.then(zeroKit => zeroKit.login(user.zeroKitId, password))
+				.then(() => this.authService.clientCredentialsLogin(user.id))
+				.then(() => {
+					sessionHandler.set('HC_User', `${user.id},${hcUserAlias}`);
+					return this.postLogin(user);
+				})
+				.then(() => ({ id: user.id, alias: hcUserAlias })),
```
Comment (Glumli) [inline src/services/ZeroKitAdapter.js]: Just foud this comma

File: src/services/ZeroKitAdapter.js (hunk 16 above)
Comment (Harsimran1) [reply src/services/ZeroKitAdapter.js]: any suggestion? createTresorAndTek? afterLogin? prepareForEncryption?

File: src/templates/loginForm.js (hunk 18 above)
Comment (Harsimran1) [reply src/templates/loginForm.js]: call? do you mean calling it as function? keeping 'document' in global scope will fail for node.

File: src/services/ZeroKitAdapter.js [hunk 20]
```diff
@@ -15,25 +16,42 @@ class ZeroKitAdapter {
...
+				document.getElementsByTagName('head')[0].appendChild(script);
 			});
-			document.getElementsByTagName('head')[0].appendChild(script);
-		});
+		}
+	}
+
+	loginNode(hcUserAlias, password) {
+		return UserService.resolveUser(hcUserAlias)
+			.then(user => this.zeroKit.then(zeroKit => zeroKit.login(user.zeroKitId, password))
```
Comment (ibihim) [inline src/services/ZeroKitAdapter.js]: Please do not nest thens... will continue review later 😊

File: src/templates/loginForm.js (hunk 18 above)
Comment (Harsimran1) [reply src/templates/loginForm.js]: getLoginForm is a function in ZeroKitAdapter. Will rename it to createLoginForm in both files

File: src/services/ZeroKitAdapter.js (hunk 20 above)
Comment (ibihim) [reply src/services/ZeroKitAdapter.js]: Ah sorry. the line length of the macbook was very limited. I didn't see the intend. Give me a second, already looking into it more closely

File: src/services/ZeroKitAdapter.js [hunk 21]
```diff
@@ -15,25 +16,42 @@ class ZeroKitAdapter {
...
-			script.addEventListener('load', () => {
-				if (zkit_sdk) {
-					zkit_sdk.setup(config.zkit.service_url, '');
-					resolve(zkit_sdk);
-				} else {
-					reject(new Error('Setting up ZeroKit failed.'));
-				}
+
+		if (NODE) {
+			this.zeroKit = require('zerokit-node-client')(config.zkit.service_url, config.zkit.version);
```
Comment (ibihim) [inline src/services/ZeroKitAdapter.js]: I have not yet complete inside into this repository, so take my opinion not without hesitation:

require is a synchronous locking call. That is the reason, why you put it at the top. Couldn't we move the require to the top, wrapped within an if clause?

E.g.
```
if (NODE) {
    import zerokitNodeClient from 'zerokit-node-client';
}
```

File: src/services/ZeroKitAdapter.js [hunk 22]
```diff
@@ -15,25 +16,42 @@ class ZeroKitAdapter {
...
+		} else {
+			this.zeroKit = new Promise((resolve, reject) => {
+				const script = document.createElement('script');
+				script.src = `${config.zkit.service_url}/static/v4/zkit-sdk.js`;
+				script.querySelector('#zkit');
+				script.addEventListener('load', () => {
+					if (zkit_sdk) {
```
Comment (ibihim) [inline src/services/ZeroKitAdapter.js]: where is zkit_sdk defined? is this some global variable?

File: src/services/ZeroKitAdapter.js (hunk 16 above)
Comment (ibihim) [reply src/services/ZeroKitAdapter.js]: aren't you calling postLogin with an argument in line 46?

File: src/services/ZeroKitAdapter.js (hunk 20 above)
Comment (ibihim) [reply src/services/ZeroKitAdapter.js]: Ok, I am fine with that approach. It is more nested, but it does not have a let user.

File: test/services/zerokitAdapterTest.js [hunk 23]
```diff
@@ -19,6 +19,8 @@ chai.use(sinonChai);
 
 const expect = chai.expect;
 
+global.NODE = false;
```
Comment (ibihim) [inline test/services/zerokitAdapterTest.js]: I would avoid setting a global variable in that way.

File: test/services/zerokitAdapterTest.js [hunk 24]
```diff
@@ -186,6 +205,40 @@ describe('zerokitAdapter', () => {
...
+				done();
+			})
+			.catch(done);
+	});
+
+	it('loginNode - rejects when wrong zerokit password is entered', (done) => {
+		loginNodeStub.rejects('error');
+
+		zerokitAdapterNode.loginNode(testVariables.userAlias, testVariables.wrongPassword)
+			.catch(() => {
```
Comment (ibihim) [inline test/services/zerokitAdapterTest.js]: it would be cool to have a then above that throws an error:
```.then(() => done(Error('This shouldn\'t happen!')))```

File: src/services/ZeroKitAdapter.js (hunk 22 above)
Comment (Harsimran1) [reply src/services/ZeroKitAdapter.js]: it is a global variable created by zerokit web sdk

File: src/services/ZeroKitAdapter.js (hunk 21 above)
Comment (Harsimran1) [reply src/services/ZeroKitAdapter.js]: unfortunately imports cannot be written inside if statements. I can however add ```let zkitNodeSDK;
if (NODE) {
	zkitNodeSDK = require('zerokit-node-client');
}``` at the top and use it in code.

File: src/services/ZeroKitAdapter.js [hunk 25]
```diff
@@ -15,20 +16,56 @@ class ZeroKitAdapter {
...
+					}
+				});
+				document.getElementsByTagName('head')[0].appendChild(script);
 			});
-			document.getElementsByTagName('head')[0].appendChild(script);
-		});
+		}
+	}
+
+	loginNode(hcUserAlias, password) {
```
[the remark this answers is not in the document]
Comment (ibihim) [reply src/services/ZeroKitAdapter.js]: I think loginNode should move closer to login / move getLoginForm above this method.

The methods are quite similar.

[the remark this answers is not in the document]
Comment (Glumli) [reply src/services/ZeroKitAdapter.js]: I already talked to simran. Wanted to push after functional review

File: src/config/develop.js [hunk 26]
```diff
@@ -11,6 +11,7 @@ const config = {
 	zkit: {
 		service_url: 'https://raf1unp2me.api.tresorit.io',
 		clientId: 'raf1unp2me_qQ1dMytsop',
+		version: 4,
```
[the remark this answers is not in the document]
Comment (Harsimran1) [reply src/config/develop.js]: Normally we sometimes need to connect to local services (which we don't do very often). And sometimes we might connect to sandbox (in case staging not working). So it makes sense to keep a separate file for development environment, rather than accidentally pushing into staging.

File: test/services/zerokitAdapterTest.js (hunk 23 above)
Comment (Harsimran1) [reply test/services/zerokitAdapterTest.js]: moved the global variable to arguments in test command

File: src/services/ZeroKitAdapter.js [hunk 27]
```diff
@@ -1,71 +1,60 @@
...
-		});
+	loginNode(hcUserAlias, password) {
+		return UserService.resolveUser(hcUserAlias)
+			.then(user => this.zeroKit.then(zeroKit => zeroKit.login(user.zeroKitId, password))
+				.then(() => this.authService.clientCredentialsLogin(user.id))
+				.then(() => {
+					sessionHandler.set('HC_User', `${user.id},${hcUserAlias}`);
+					return this.setupUser();
+				})
+				.then(() => ({ id: user.id, alias: hcUserAlias })),
```
Comment (Glumli) [inline src/services/ZeroKitAdapter.js]: Another Comma

Comment (Harsimran1) [reply src/services/ZeroKitAdapter.js]: eslint shows error without it

Comment (Glumli) [reply src/services/ZeroKitAdapter.js]: I would remove the comma and remove the linebreak

File: src/services/ZeroKitAdapter.js [hunk 28]
```diff
@@ -49,8 +49,7 @@ class ZeroKitAdapter {
 					sessionHandler.set('HC_User', `${user.id},${hcUserAlias}`);
 					return this.setupUser();
 				})
-				.then(() => ({ id: user.id, alias: hcUserAlias })),
-			);
+				.then(() => ({ id: user.id, alias: hcUserAlias })));
```
Comment (Glumli) [inline src/services/ZeroKitAdapter.js]: SO PRETTY

File: src/services/ZeroKitAdapter.js [hunk 29]
```diff
@@ -108,6 +134,7 @@ class ZeroKitAdapter {
 
 	getRegistrationForm(parentElement) {
 		return new Promise((resolve, reject) => {
+			const registrationForm = createRegistrationForm();
```
Comment (isalga) [inline src/services/ZeroKitAdapter.js]: `registrationFormTemplate` would make code clearer.

File: src/services/ZeroKitAdapter.js [hunk 30]
```diff
@@ -95,10 +89,42 @@ class ZeroKitAdapter {
 							UserService.user.tek = createdTek;
 						});
 				}
-				return { id: userId, alias: hcUserAlias };
 			});
 	}
 
+	getLoginForm(parentElement) {
+		return new Promise((resolve, reject) => {
+			const loginForm = createLoginForm();
```
Comment (isalga) [inline src/services/ZeroKitAdapter.js]: `loginFormTemplate` would make code clearer.

File: src/templates/loginForm.js [hunk 31]
```diff
@@ -4,20 +4,23 @@ const tagIds = {
...
-	</div>
-	<div class="row">
-		<label>Passwort</label>
-	  <div id="${tagIds.zkitLogin}"/></div>
-	</div>
-	<div class="row">
-  	<button id="${tagIds.submitBtn}">ANMELDEN</button>
-	</div>
-`;
+const createLoginForm = () => {
```
Comment (isalga) [inline src/templates/loginForm.js]: same, I would prefer `getLoginFormTemplate` or `createLoginFormTemplate`

File: src/templates/registrationForm.js [hunk 32]
```diff
@@ -3,17 +3,20 @@ const tagIds = {
...
-	<div class="row">
-		<label>E-Mail-Adresse</label>
-  	<input type="text" id="${tagIds.hcUserRegister}" placeholder="Email">
-	</div>
-	<div class="row">
-		<label>Passwort</label>
-  	<div id="${tagIds.zkitRegistration}"></div>
-	</div>
-`;
+const createRegistrationForm = () => {
```
Comment (isalga) [inline src/templates/registrationForm.js]: same

File: package.json [hunk 33]
```diff
@@ -21,7 +25,8 @@
     "es6-promise": "^4.1.0",
     "json-server": "^0.10.3",
     "superagent": "^3.5.2",
-    "superagent-bluebird-promise": "^4.2.0"
+    "superagent-bluebird-promise": "^4.2.0",
+    "zerokit-node-client": "^4.0.0"
```
Comment (isalga) [inline package.json]: Coolio

File: src/HealthCloud.js [hunk 34]
```diff
@@ -29,6 +25,13 @@ class HealthCloud {
 		this.grantPermission = zeroKitAdapter.grantPermission.bind(zeroKitAdapter);
 		this.getGrantedPermissions = UserService.getGrantedPermissions.bind(UserService);
 		this.logout = zeroKitAdapter.logout.bind(zeroKitAdapter);
+		if (NODE) {
```
Comment (isalga) [inline src/HealthCloud.js]: Is there not a way to make this NODE especialitzation via "constructor" parameter during instantiation. Something like a object factory.

File: src/HealthCloud.js [hunk 35]
```diff
@@ -37,4 +29,26 @@ class HealthCloud {
...
+	constructor(options) {
+		const authService = new AuthService(options);
+		const zeroKitAdapter = new ZeroKitAdapter({ authService });
+		const documentService = new DocumentService({ zeroKitAdapter });
+		super({ documentService, zeroKitAdapter });
+		this.login = zeroKitAdapter.loginNode.bind(zeroKitAdapter);
+	}
+}
+
+module.exports = NODE ? HealthCloudNode : HealthCloudWeb;
```
Comment (isalga) [inline src/HealthCloud.js]: Very coolio

File: test/services/zerokitAdapterTest.js (hunk 24 above)
Comment (ibihim) [reply test/services/zerokitAdapterTest.js]: seems to be done 😄

File: test/services/zerokitAdapterTest.js [hunk 36]
```diff
@@ -267,13 +319,18 @@ describe('zerokitAdapter', () => {
 	it('getLoginForm - Happy Path', (done) => {
 		const parentElement = { appendChild: sinon.stub() };
 
+		const createdLoginForm = loginForm.createLoginFormTemplate();
+		let createLoginFormStub = sinon.stub(loginForm, 'createLoginFormTemplate').returns(createdLoginForm);
```
Comment (ibihim) [inline test/services/zerokitAdapterTest.js]: why not const here?

File: src/services/AuthService.js [hunk 37]
```diff
@@ -74,6 +75,19 @@ class Auth {
...
+		const body = {
+			client_id: this.clientId,
+			grant_type: 'client_credentials',
+			client_secret: this.secret,
+			scope: `user:${userId}`,
+		};
+		return authRoutes.getAccessTokenFromCredentials(body)
+			.then((res) => {
+				sessionHandler.set('HC_Auth', res.access_token);
+			});
```
Comment (ibihim) [inline src/services/AuthService.js]: I followed this path up... and as it seems we do not catch any errors for this...

I think @Glumli mentioned something about fixing error handling in the SDK, do we have a story for that?

File: src/services/ZeroKitAdapter.js [hunk 38]
```diff
@@ -1,71 +1,59 @@
...
-			script.querySelector('#zkit');
-			script.addEventListener('load', () => {
-				if (zkit_sdk) {
-					zkit_sdk.setup(config.zkit.service_url, '');
-					resolve(zkit_sdk);
-				} else {
-					reject(new Error('Setting up ZeroKit failed.'));
-				}
+
+		if (NODE) {
```
Comment (ibihim) [inline src/services/ZeroKitAdapter.js]: Isn't there a way to omit the if (NODE) solution by some smart comments and webpack magic? 
e.g. 
```JavaScript
/* webpack X command only in NODE_ENV=true START */
{
  /* Node Code */
}
/* webpack X command only in NODE_ENV=true END*/
```

Comment (ibihim) [reply src/services/ZeroKitAdapter.js]: It is not a request change! 😄

File: src/services/AuthService.js (hunk 37 above)
Comment (ibihim) [reply src/services/AuthService.js]: Seems to be a general issue...

File: test/routes/authRoutesTest.js [hunk 39]
```diff
@@ -45,6 +66,45 @@ describe('user Routes', () => {
 		});
 	});
 
+	it('getAccessTokenFromCredentials passes', (done) => {
```
[the remark this answers is not in the document]
Comment (Harsimran1) [reply test/routes/authRoutesTest.js]: The PR will become a lot bigger if we fix these here. Created a story https://ares.epic.hpi.uni-potsdam.de/epicjira/jira/browse/HC-839

File: src/services/AuthService.js (hunk 37 above)
Comment (Harsimran1) [reply src/services/AuthService.js]: there is another PR for using local storage in web for sessionHandler, but we still need a discuss and create a general ticket about handling set case
